testplan/testing/filtering.py

```diff
@@ -285,10 +293,61 @@ def parse_pattern(self, pattern: str) -> List[str]:
                 )
             )
 
-        return patterns + ([self.ALL_MATCH] * (self.MAX_LEVEL - len(patterns)))
+        test_level, suite_level, case_level = patterns + (
+            [self.ALL_MATCH] * (self.MAX_LEVEL - len(patterns))
+        )
 
-    def filter_test(self, test):
-        return fnmatch.fnmatch(test.name, self.test_pattern)
+        m = re.match(TEST_PART_REGEX, test_level)
```

Comment:
I have two problem here, one is that the pattern match means limitation in the naming of multitests. 

The information currently is not available from theoutput of info:

```
python.exe .\test_plan.py --info pattern-full
MultitestParts
  MultitestParts::Suite1
    MultitestParts::Suite1::test_equal
    MultitestParts::Suite1::test_greater
  MultitestParts::Suite2
    MultitestParts::Suite2::test_bool <val=1>
    MultitestParts::Suite2::test_bool <val=4>
MultitestParts
  MultitestParts::Suite1
    MultitestParts::Suite1::test_not_equal
    MultitestParts::Suite1::test_approximate_equal
  MultitestParts::Suite2
    MultitestParts::Suite2::test_bool <val=2>
    MultitestParts::Suite2::test_bool <val=5>
MultitestParts
  MultitestParts::Suite1
    MultitestParts::Suite1::test_less
  MultitestParts::Suite2
    MultitestParts::Suite2::test_bool <val=0>
    MultitestParts::Suite2::test_bool <val=3>
Exiting code: 0
```

if one want to run just all from the 2nd part they need to count how many parts are there, and hoping the order is just correct in the output. If there is a problem in calculation they got nothing

```
python.exe ./test_plan.py --patterns "MultitestParts - part(1/2):*:*" --info pattern-full
Exiting code: 0
```

This also makes(keeps) multiple entry in the json lister for a single Multitest and anyone try to derive ids could not doo it from there. Also the --info is not in sync with ids in the result json and results xml, so one (a software) could not easily correlate betwing list of testcases from cli , the pattern need to be provided and the results it is generating.

If we would be just listing the uid or full name whatever is the best to call it, then patterns still can nicely works with `"MultitestParts*:Suite1:*` way. That looks simple in testplan (no restriction on naming, no parse name for filtering), simple the user to interpret and work with, fit nice for automation on listing <--> filtering <--> result crossreference.

Reply:
Sure I will definitely make the `--list` output conform with input patterns, will also check `--patterns` using together with `--info`

Tbh I don't like the single asterisk way. It can easily include other tests happening to have the same name prefix. It cannot express whether the user only wants to select cases from a specific part or from all the parts.

Reply:
I see your point on the single asterix case, however if that is a problem for a project they can, add the part as well to the pattern just use aterix in place of a the part numbers, I do nto expect a collide there, but if tehre are they still can change their naming, or go with multiple patterns to capture all. Figuring out a the right pattern if you do not see it for a part is not good.

Reply:
I want to support the use case that user want to select some specific testcases with patterns, regarding less of parts. While with the auto parts feature, there could be cases that parted multitest no longer getting parted in another run.